crates/e2e/src/setup/solver/mock.rs

```diff
@@ -15,32 +15,54 @@ use {
 /// A solver that does not implement any solving logic itself and instead simply
 /// forwards a single hardcoded solution.
 pub struct Mock {
-    /// The currently configured solution to return.
-    solution: Arc<Mutex<Option<Solution>>>,
+    /// Mock solver shareable state
+    state: State,
     /// Under which URL the solver is reachable by a driver.
     pub url: Url,
 }
 
+#[derive(Clone)]
+pub struct State {
+    /// In-memory set of the auctions received by the solver.
+    auctions: Arc<Mutex<Vec<Auction>>>,
+    /// The currently configured solution to return.
+    solution: Arc<Mutex<Option<Solution>>>,
+}
+
 impl Mock {
     /// Instructs the solver to return a new solution from now on.
     pub fn configure_solution(&self, solution: Option<Solution>) {
-        *self.solution.lock().unwrap() = solution;
+        *self.state.solution.lock().unwrap() = solution;
+    }
+
+    /// Returns all the auctions received by the solver
+    pub fn get_auctions(&self) -> Vec<Auction> {
```

Comment:
I think it's fine to return a reference here to avoid unnecessary clones. That a mutex guard can be held too long is okay in my books since this is an error that should be pretty obvious from failing tests.